extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java

```diff
@@ -0,0 +1,459 @@
+            return response.statusCode();
+        }
+
+        @Override
+        protected Long responseContentLength(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        protected Long responseContentLengthUncompressed(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        protected List<String> responseHeader(final HttpRequest request, final HttpResponse response, final String name) {
+            return response.headers().getAll(name);
+        }
+    }
+
+    // TODO - Ideally this should use HttpSpanNameExtractor, but to keep the name without the slash we use our own.
+    private static class ServerSpanNameExtractor implements SpanNameExtractor<HttpRequest> {
+        private final HttpServerAttributesExtractor<HttpRequest, HttpResponse> serverAttributesExtractor;
+
+        ServerSpanNameExtractor(
+                final HttpServerAttributesExtractor<HttpRequest, HttpResponse> serverAttributesExtractor) {
+            this.serverAttributesExtractor = serverAttributesExtractor;
+        }
+
+        @Override
+        public String extract(final HttpRequest httpRequest) {
+            if (httpRequest instanceof HttpServerRequest) {
+                String path = URI.create(httpRequest.uri()).getPath();
+                if (path != null && path.length() > 1) {
+                    return path.substring(1);
+                } else {
+                    return "HTTP " + httpRequest.method();
+                }
+            }
+            return null;
+        }
+
+        static SpanNameExtractor<HttpRequest> create(
+                HttpServerAttributesExtractor<HttpRequest, HttpResponse> serverAttributesExtractor) {
+            return new ServerSpanNameExtractor(serverAttributesExtractor);
+        }
+    }
+
+    private static class AdditionalServerAttributesExtractor implements AttributesExtractor<HttpRequest, HttpResponse> {
+        @Override
+        public void onStart(final AttributesBuilder attributes, final HttpRequest httpRequest) {
+            if (httpRequest instanceof HttpServerRequest) {
+                set(attributes, HTTP_CLIENT_IP, VertxUtil.extractClientIP((HttpServerRequest) httpRequest));
+            }
+        }
+
+        @Override
+        public void onEnd(
+                final AttributesBuilder attributes,
+                final HttpRequest httpRequest,
+                final HttpResponse httpResponse,
+                final Throwable error) {
+
+            // The UrlPathTemplate is only added to the Vert.x context after the instrumenter start
+            if (httpRequest instanceof HttpRequestSpan) {
+                HttpRequestSpan httpRequestSpan = (HttpRequestSpan) httpRequest;
+                // RESTEasy
+                String route = httpRequestSpan.getContext().getLocal("UrlPathTemplate");
```

Comment:
You take a risk here. 
That operation is only allowed on duplicated context. In this case, it should be fine, as it should dispatch the method on a duplicated context. 

But, I can quickly imagine code where it switches to a root context. In this case, this would not work anymore. So, the question is: should it be more defensive?